apps/mymedicare_cb/views.py

```diff
@@ -62,16 +50,50 @@ def callback(request):
         response.raise_for_status()
     except requests.exceptions.HTTPError as e:
         logger.error("User info request response error {reason}".format(reason=e))
-        return JsonResponse({
-            "error": 'An error occurred connecting to account.mymedicare.gov'
-        }, status=502)
+        raise UpstreamServerException(
+            'An error occurred connecting to account.mymedicare.gov')
 
     # Get the userinfo response object
     user_info = response.json()
     user = get_and_update_user(user_info)
-    login(request, user)
+    request.user = user
```

Comment:
For my own edification -- this is the real magic here right? By avoiding `login()` we skip session creation?

Reply:
right and then we use the `Approval` later as what is essentially a replacement for a session